extensions/kind/package.json

```diff
@@ -48,11 +48,24 @@
           "scope": "KubernetesProviderConnectionFactory",
           "description": "Setup an ingress controller (Contour https://projectcontour.io)"
         },
+        "kind.cluster.creation.switchControlPlaneImage": {
+          "type": "boolean",
+          "default": false,
+          "scope": "KubernetesProviderConnectionFactory",
+          "description": "Using custom node’s container image"
+        },
         "kind.cluster.creation.controlPlaneImage": {
           "type": "string",
           "default": "",
           "scope": "KubernetesProviderConnectionFactory",
-          "placeholder": "Leave empty for using latest.",
+          "hidden": {
+            "key": "kind.cluster.creation.switchControlPlaneImage",
+            "operator": "NotEqual",
+            "values": [
+              "true"
+            ]
+          },
+          "placeholder": "kindest/node:v1.27.3@sha256:...",
```

Comment:
hello @axel7083 and thanks for your pull request

I would rather not introduce a new conditional system.

we already have a context-key implementation that allow to use "natural language" like `!isLinux() && myProperty`

full list here

the when clause is used to display the element, not to hide it
https://code.visualstudio.com/api/references/when-clause-contexts#conditional-operators

Reply:
So would you prefer to adapt the same mechanism used for the when inside the hide ? Or abandon the idea of conditional field ?

Reply:
hi @axel7083 

I would say the `when` clause is already supporting the conditional. If the when condition is false the property is not displayed so this is why I'm a bit lost when you're saying we should abandon the idea of conditional field as for me 'when' is a conditional clause (that can reference one , multiple properties and multiple conditions)

Reply:
> hi @axel7083
> 
> I would say the `when` clause is already supporting the conditional. If the when condition is false the property is not displayed so this is why I'm a bit lost when you're saying we should abandon the idea of conditional field as for me 'when' is a conditional clause (that can reference one , multiple properties and multiple conditions)

Thanks for your reply ! Yeah sorry I wasn't expressing myself properly, I meant abandoning/closing this PR ! 

I am looking to know if the when is meant for supporting other context such as the current form it is in ? Currently the context does seems to have much element in it, that what is losing me about it 

Also in the current implementation the function filtering on the when condition is inside the onMount therefore is not called when the form have value updated.

Moreover, I am a bit surprised of how the form is actually updating the field, the method `setRecordValue` defined here :

https://github.com/containers/podman-desktop/blob/7991d543e13d6d3171fa6a9c95b867a9776d2ac3/packages/renderer/src/lib/preferences/PreferencesRenderingItemFormat.svelte#L20

is not called for all types, which is a bit confusing, some field are updated through it, other does not seems to be (?).

Reply:
so yes maybe there are enhancements to do to update it live (and not only at creation/first display time) but in the when clause you can already express your need

then, I would focus on having the when clause used in the Wizard tacking care of tracking changes of the expressions/contexts to hide/display items live

Reply:
Okey thanks for the direction, I will take a look !